sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3062,9 +3068,12 @@ def _update_after_cluster_provisioned(
             resources_utils.port_ranges_to_set(current_ports) -
             resources_utils.port_ranges_to_set(prev_ports))
         if open_new_ports:
-            with rich_utils.safe_status(
-                    '[bold cyan]Launching - Opening new ports'):
-                self._open_ports(handle)
+            cloud = handle.launched_resources.cloud
+            if not (cloud.OPEN_PORTS_VERSION <=
+                    clouds.OpenPortsVersion.OPEN_ON_LAUNCH_ONLY):
+                with rich_utils.safe_status(
+                        '[bold cyan]Launching - Opening new ports'):
+                    self._open_ports(handle)
```

Comment:
Currently, we have some logic to calculate the newly-required ports (e.g. launched with `8000-8010`, now adding `8005-8015`, only need to add `8011-8015`). If we want to move the entire logic into `bulk_provision`, we need to move that logic as well. I'm slightly concerned if it will be the best place to put that logic but open to discussions.

Also, this will be a big refactoring for clouds that create SG and update them later - now we need to create/update a fully configured SG in the first place. Do you think we could do this in another PR?